soc/nxp/imx/imx7d/linker.ld

```diff
@@ -0,0 +1,17 @@
+/*
```

Comment:
Please, see the soc/nxp/imx structure: https://github.com/zephyrproject-rtos/zephyr/tree/main/soc/nxp/imx/imx8m
This linker script is for m4 core so I suggest to add this file in an m4 folder.

Otherwise, TLGM.

Reply:
Thanks for your review, 

I suggest that another PR is provided later on to properly refactor the imx7d folder after this one is merged.
This is because the soc/soc_clk_freq files are all at the root of imx7d and I'd like to avoid introducing too many changes here.

Can you confirm that the A7 core is supported and tested on Zephyr for the IMX7D ? Because it might not be relevant for this PR to handle refactoring the folder structure of imx7d soc.

I do not plan personally on providing support for the A7 core but if someone is interested at some point, I suggest they take over the folder refactoring task :-) 

How would you suggest proceeding?

Reply:
The imx7d seems to support only m4 core. 
So, leave it like this.. for now.
Thanks!